File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0 || nums.size() == 1){
            return nums.size();
        }
        unordered_set <int> st(nums.begin(), nums.end());
        int ans = 0;
        for (auto i : st) {
            if (st.find(i - 1) == st.end()) {
                int length = 1;
                while (st.find(i + length) != st.end()) {
                    length++;
                }
                ans = max(ans, length);
            }
        }
        return ans;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        ans = 0
        hashset = set(nums)

        for num in hashset:
            if num - 1 not in hashset:
                curr_length = 1
                while num + 1 in hashset:
                    curr_length += 1
                    num += 1
                ans = max(ans, curr_length)

        return ans